test/spec/hooks/useStickyIntersectionObserver.spec.js

```diff
@@ -113,14 +140,30 @@ describe('hooks/userStickyIntersectionObserver', function() {
 
 function noop() {}
 
+/**
+ * Overrides the IntersectionObserver global with a mock.
+ *
+ * @param {Object} props
+ * @param {Object} [props.observe]
+ * @param {Object} [props.unobserve]
+ * @returns {Function} triggers the callback on all created observers
+ */
 function mockIntersectionObserver(props) {
   const {
     observe = noop,
     unobserve = noop
   } = props;
 
-  global.IntersectionObserver = class IntersectionObserver {
-    constructor() {}
+  const callbacks = [];
+
+  function triggerCallbacks(args) {
+    callbacks.forEach(callback => callback(args));
+  }
+
+  const MockObserver = class IntersectionObserver {
```

Comment:
`extends` is necessary if you want the mock to have IntersectionObserver's APIs.

```suggestion
  const MockObserver = class extends IntersectionObserver {
```

Reply:
I don't think we want to extend it, rather replace it with our own implementation. But you are right that this looks wrong an could just be
```suggestion
  class MockObserver {
```